internal/matchers/ogg.go

```diff
@@ -0,0 +1,54 @@
+package matchers
+
+import (
+	"bytes"
+)
+
+/*
+NOTE:
+
+In May 2003, two Internet RFCs were published relating to the format.
+The Ogg bitstream was defined in RFC 3533 (which is classified as
+'informative') and its Internet content type (application/ogg) in RFC
+3534 (which is, as of 2006, a proposed standard protocol). In
+September 2008, RFC 3534 was obsoleted by RFC 5334, which added
+content types video/ogg, audio/ogg and filename extensions .ogx, .ogv,
+.oga, .spx.
+
+See:
+https://tools.ietf.org/html/rfc3533
+https://developer.mozilla.org/en-US/docs/Web/HTTP/Configuring_servers_for_Ogg_media#Serve_media_with_the_correct_MIME_type
+https://github.com/file/file/blob/master/magic/Magdir/vorbis
+*/
+
+// Ogg matches an Ogg file.
+func Ogg(in []byte) bool {
+	return len(in) > 5 && bytes.Equal(in[:5], []byte("\x4F\x67\x67\x53\x00"))
+}
+
+// OggAudio matches an audio ogg file.
+func OggAudio(in []byte) bool {
+
+	if !Ogg(in) {
```

Comment:
Because `oggAudio` is a child node of `ogg`, this call to `Ogg(in)` is always guaranteed to return true. I would remove this if clause. Is there anything I'm missing?

Reply:
Oops, I overlooked the fact that although `OggAudio` is an exported function, the package is internal.

The if clause is required if `OggAudio` and `OggVideo` is or will be part of the public API (outside of the hierarchical detection model). Both Ogg audio and video files must first qualify as Ogg container format. Hence, instead of duplicating the code to check if it's an Ogg container, I invoked the function that does the check.

Let me know if the if clause needs to be removed.

Reply:
That's right! `mimetype` is the only package to call these functions and the call should be removed.

Reply:
Thanks! Done 👍